### PR TITLE
1040 make dataviews reactive

### LINKED ISSUE #1040: Make dataviews reactive to attrs changes
Ditto https://github.com/CartoDB/deep-insights.js/issues/117

When dataview model have its attrs changed they're not reflected on data rendered, seems like.

### PR DESCRIPTION
**Pull request by viddo**
Fixes #1040 

Basically, renamed `_onFilterChanged ` to `_reloadMap` (since that's what it does), and use that as the side-effect on attrs changes.

@alonsogarciapablo review, please 
cc @javisantana

### CODE REVIEW

**Comment by javisantana** [comment]
@alonsogarciapablo a question: does it make sense to have a reloadMap in the new architecture?

File: src/dataviews/dataview-model-base.js [hunk 1]
```diff
@@ -60,8 +60,19 @@ module.exports = Model.extend({
 
     // Retrigger an event when the filter changes
     if (this.filter) {
-      this.listenTo(this.filter, 'change', this._onFilterChanged);
+      this.listenTo(this.filter, 'change', this._reloadMap);
     }
+
+    this.bind('change:operation', this._reloadMap, this);
```
Comment (alonsogarciapablo) [inline src/dataviews/dataview-model-base.js]: I don't think we should leave this here since not all types of dataviews have an `operation` attribute.

**Comment by alonsogarciapablo** [comment]
@javisantana @viddo here's how `_onFilterChanged` works on the branch that implements filtering in the client:

```
  _onFilterChanged: function (filter) {
    var dataProvider = this.layer.getDataProvider();
    if (dataProvider) {
      dataProvider.applyFilter(this.get('column'), filter);
    } else {
      this._map.reload({
        sourceLayerId: this.layer.get('id')
      });
    }
  },
```

It will make sense to keep `_onFilterChange` IMO, so perhaps we can have the following in order to avoid conflicts:

```
initialize: function () {
  ...
  this.listenTo(this.filter, 'change', this._onFilterChanged);
  this.bind('change:operation', this._reloadMap, this);
  ...
},

_onFilterChanged: function () {
  this._reloadMap();
},

_reloadMap: function () {
  map.reload(...);
},
```

File: src/dataviews/dataview-model-base.js (hunk 1 above)
Comment (viddo) [inline src/dataviews/dataview-model-base.js]: Yes, you're right, I'll fix this one

**Pull request merged**